checker/src/org/checkerframework/checker/guieffect/GuiEffectTypeFactory.java

```diff
@@ -432,6 +509,14 @@ protected TreeAnnotator createTreeAnnotator() {
         return new ListTreeAnnotator(super.createTreeAnnotator(), new GuiEffectTreeAnnotator());
     }
 
+    public void constrainLambdaToUI(LambdaExpressionTree lambdaExpressionTree) {
```

Comment:
Is the commented code below not supposed to persist the type for that Tree node?

Reply:
@wmdietl